Add: Artist Search Feature

sawyerf opened the pull request:
Hi,

I added an artist search feature and made some UI adjustments to support it.

![Screenshot 2025-02-12 at 00-31-24 Good Music - Good Feeling](https://github.com/user-attachments/assets/c96a8f07-a709-43ad-9946-fcf3d8ce9db7)

![Screenshot 2025-02-12 at 00-32-40 Good Music - Good Feeling](https://github.com/user-attachments/assets/cad3d1a7-45ab-462f-9216-0e044ac493df)


Note: I didn’t dare to re-indent everything, but HTML and JS are not properly indented.

kmille commented:
Hey,
do you know why I don't get the `top tracks` of an artist? Like when I'm looking for `Adele`
https://www.deezer.com/us/artist/75798/top_track

The UI confuses me a bit. Not sure if I want to merge this to be honest (tradoff UI bloat vs useful feature - I like it to be kISS)

sawyerf commented:
I get all the albums of an artist, not the top track.

What confuses you?

sawyerf commented:
What confuses you about the UI is it the design or the logic ? Personally, I find it much more intuitive to have a single search button with category selection buttons. That’s what all other websites do; no one chooses to have a separate search button for each category.

![Example google](https://github.com/user-attachments/assets/e2f575c7-dfcb-4737-a7be-296c87b90964)

![Example deezer](https://github.com/user-attachments/assets/06855b76-eca9-42f0-8c00-724fed99c986)

sawyerf commented:
If you prefer, I made this design that I can push:

![Other style](https://github.com/user-attachments/assets/c2c4e899-278c-4aa9-ab66-d1df492ca392)

kmille commented:
I dont know when I have time to review this. So please be patient. 

On February 13, 2025 7:05:49 PM GMT+01:00, sawyerf ***@***.***> wrote:
>sawyerf left a comment (kmille/deezer-downloader#137)
>
>If you prefer, I made this design that I can push:
>
>![Other style](https://github.com/user-attachments/assets/c2c4e899-278c-4aa9-ab66-d1df492ca392)
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/kmille/deezer-downloader/pull/137#issuecomment-2657367030
>You are receiving this because you commented.
>
>Message ID: ***@***.***>

sawyerf commented:
ping

Yann-Miquel-Erdmann commented:
I would like this feature too.

vanhauser-thc commented:
yes please add this feature

kmille commented:
Hey,
just had some time looking at your MR. Can you please
- rename Songs to Tracks in the UI? Then it's the same wording Deezer uses
- rename Artist to Artists in the UI?
- fix this: search for artist => click on "list artist albums" => backend gives 500 (
```
  File "/home/kmille/projects/deezer-downloader/deezer_downloader/deezer.py", line 472, in deezer_search
    for item in data:
UnboundLocalError: cannot access local variable 'data' where it is not associated with a value

```
- resolve the merge conflict
- feature request: "Top tracks of an artist" (https://www.deezer.com/en/artist/75798/top_track). Can you add such button to the artist search results? There is currently a button 'list artist albums'. That would be really nice. We don't cover this right now, but I think it's a nice feature to say "Give me the best/top songs of artist X"

### Some smaller things
- When I click on search when the search text box is empty: do not send a request. Just do nothing instead. Right now there is a 400 response and nothing is happening in the UI
- There is still the help text "Alt+Enter: search albums" in the footer. As this is not the case anymore. Can remove this?

After checking the UI/functionality, I will review the code. I will also squash the commits.

Thanks for your contribution! I actually really like it spending some time with it.

sawyerf commented:
Hi !
Thanks for review my PR ! :)
I fixed all the problems, I added the artist's top tracks button and I also deleted the cover column for album tracks table.

![a](https://github.com/user-attachments/assets/33d58837-ec94-4ae7-9cd0-f3719ba34170)

- [x]  rename Songs to Tracks in the UI? Then it's the same wording Deezer uses
- [x]  rename Artist to Artists in the UI?
- [x]  fix this: search for artist => click on "list artist albums" => backend gives 500
- [x]  resolve the merge conflict
- [x]  feature request: "Top tracks of an artist" (https://www.deezer.com/en/artist/75798/top_track).

Some smaller things

- [x]  When I click on search when the search text box is empty: do not send a request. Just do nothing instead. Right now there is a 400 response and nothing is happening in the UI
- [x]  There is still the help text "Alt+Enter: search albums" in the footer. As this is not the case anymore. Can remove this?

kmille commented:
Looks good! Some smaller things:

- Can you please run `poetry update` to update the deps (or use the file from master branch). Right now we have a conflict
- There is also: `TYPE_PLAYLIST = "playlist"`, which is not used. Do you know if it's possible to search for playlists?

I really like how it looks like on mobile devices!

kmille commented on deezer_downloader/deezer.py:
```diff
@@ -450,13 +453,23 @@ def deezer_search(search, search_type):
         return []
     search = urllib.parse.quote_plus(search)
     try:
         if search_type == TYPE_ALBUM_TRACK:
             data = get_song_infos_from_deezer_website(TYPE_ALBUM, search)
```
Do you know why we still use `get_song_infos_from_deezer_website`? There is a check if we are logged in (If `MD5_ORIGIN` exists in the response). Would be nice to just the API for that `TYPE_ALBUM_TRACK:` use case, too.

sawyerf replied:
no idea there is a api link https://api.deezer.com/album/${id}

kmille replied:
thanks, I removed the old code path

sawyerf commented:
- Are you sure there is a conflict ? I never touch poetry.lock it will create conflict to run poetry update
- I find this api endpoint https://api.deezer.com/search/playlist?q=limsa but maybe in another PR

kmille commented:
![2025-05-18-153532_screenshot](https://github.com/user-attachments/assets/2566c41f-6fd6-4263-92d2-23b65b6857ad)
I will handle this. I want to change some smaller things and will

kmille commented:
Hey folks, can you please test? Then I will release an update!
I think we are good :tada:

sawyerf commented:
It work for me !

kmille commented:
Published alpha release
Docker: kmille2/deezer-downloader:v2.6.1a0-alpine3.21
pip: https://pypi.org/project/deezer-downloader/2.6.1a0/
